File: src/main/java/pl/yalgrin/playnite/simplesync/web/objects/FilterPresetResource.java
package pl.yalgrin.playnite.simplesync.web.objects;

import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;
import pl.yalgrin.playnite.simplesync.dto.objects.FilterPresetDTO;
import pl.yalgrin.playnite.simplesync.helper.SingleExecutorHelper;
import pl.yalgrin.playnite.simplesync.service.objects.FilterPresetService;
import reactor.core.publisher.Mono;

@RestController
@RequestMapping("/api/filter-preset")
@RequiredArgsConstructor
public class FilterPresetResource {
    private final FilterPresetService service;
    private final SingleExecutorHelper singleExecutorHelper;

    @GetMapping("/{id}")
    public Mono<FilterPresetDTO> getFilterPreset(@PathVariable Long id) {
        return service.findById(id);
    }

    @PostMapping("/save")
    public Mono<FilterPresetDTO> saveFilterPreset(@RequestBody FilterPresetDTO dto, @RequestParam String clientId) {
        return singleExecutorHelper.runOnExecutor(service.saveObject(dto, clientId));
    }

    @PostMapping("/delete")
    public Mono<Void> deleteFilterPreset(@RequestBody FilterPresetDTO dto, @RequestParam String clientId) {
        return singleExecutorHelper.runOnExecutor(service.deleteObject(dto, clientId));
    }
}

File: src/main/kotlin/pl/yalgrin/playnite/simplesync/web/library/FilterPresetResource.kt
package pl.yalgrin.playnite.simplesync.web.library

import org.springframework.web.bind.annotation.*
import pl.yalgrin.playnite.simplesync.helper.SingleExecutorHelper
import pl.yalgrin.playnite.simplesync.library.dto.FilterPresetDTO
import pl.yalgrin.playnite.simplesync.service.objects.FilterPresetService
import reactor.core.publisher.Mono

@RestController
@RequestMapping("/api/filter-preset")
class FilterPresetResource(
    private val service: FilterPresetService,
    private val singleExecutorHelper: SingleExecutorHelper
) {

    @GetMapping("/{id}")
    fun getFilterPreset(@PathVariable id: Long): Mono<FilterPresetDTO> {
        return service.findById(id)
    }

    @PostMapping("/save")
    fun saveFilterPreset(@RequestBody dto: FilterPresetDTO): Mono<FilterPresetDTO> {
        return singleExecutorHelper.runOnExecutor(service.saveObject(dto))
    }

    @PostMapping("/delete")
    fun deleteFilterPreset(@RequestBody dto: FilterPresetDTO): Mono<Void> {
        return singleExecutorHelper.runOnExecutor(service.deleteObject(dto))
    }
}